File: backend/account/services/parent_display.py
"""Shared formatting for orang tua / keluarga fields when deceased (almarhum)."""

ALM_PREFIX = "Alm."
ALM_AGE_PLACEHOLDER = "-"


def format_parent_name(name: str | None, *, almarhum: bool) -> str:
    """Return display name; prepend ``Alm.`` when marked almarhum."""
    raw = (name or "").strip()
    if not almarhum:
        return raw
    if not raw:
        return ALM_PREFIX
    upper = raw.upper()
    if upper.startswith("ALM.") or upper.startswith("ALM "):
        return raw
    return f"{ALM_PREFIX} {raw}"


def format_parent_age(age, *, almarhum: bool, fallback: str = "") -> str:
    """Return age for display; ``-`` when marked almarhum."""
    if almarhum:
        return ALM_AGE_PLACEHOLDER
    if age is None:
        return fallback
    s = str(age).strip()
    return s if s else fallback

File: frontend/src/lib/parent-display.ts
/** Display formatting for orang tua when marked almarhum. */

export function formatParentName(
  name: string | null | undefined,
  almarhum?: boolean,
): string {
  const raw = (name ?? "").trim()
  if (!almarhum) return raw
  if (!raw) return "Alm."
  const upper = raw.toUpperCase()
  if (upper.startsWith("ALM.") || upper.startsWith("ALM ")) return raw
  return `Alm. ${raw}`
}

export function formatParentAge(
  age: number | null | undefined,
  almarhum?: boolean,
): string | number | null | undefined {
  if (almarhum) return "-"
  return age
}
